### PR TITLE
BCDA-4681: Update ALR FHIR Resource 

### PR DESCRIPTION
**Pull request by katyasoup**
### Fixes [BCDA-4681](https://jira.cms.gov/browse/BCDA-4681)
We have been incrementally updating our mapping of ALR data to STU3 and R4 FHIR resources. This PR brings in data from ALR Table 1-7 (Covid19-related data), as well as additional fields from Tables 1-5 and 1-6 (additional assigned/assignable beneficiary information).

### Proposed Changes
- Adds a new `covidEpisode` resource to the v1 and v2 AlrBulkFhir models
- Adds additional fields to the `group` resource from tables 1-5 and 1-6
- Updates synthetic data generator and ALR tests to include new `covidEpisode` resource and additional `group` resource fields

### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
Synthetic data generated should match the structure of example data in the ALR IG
CI/CD passes

### Feedback Requested
This is a work in progress; please feel free to suggest changes or alternative approaches, especially for the extension mappings in `covidepisode.go`

### CODE REVIEW

File: bcdaworker/worker/alr_test.go [hunk 1]
```diff
@@ -119,9 +119,12 @@ func (s *AlrWorkerTestSuite) TestProcessAlrJob() {
...
 	assert.NoError(s.T(), err)
-    err = s.alrWorker.ProcessAlrJob(ctx, s.jobArgs[1])
-    // Check Job is processed with no errors
-    assert.NoError(s.T(), err)
+	err = s.alrWorker.ProcessAlrJob(ctx, s.jobArgs[1])
+	// Check Job is processed with no errors
+	assert.NoError(s.T(), err)
+	err = s.alrWorker.ProcessAlrJob(ctx, s.jobArgs[1])
+	// Check Job is processed with no errors
+	assert.NoError(s.T(), err)
```
Comment (jphechter) [inline bcdaworker/worker/alr_test.go:127]: It looks like lines `122` and `125` are referencing the same index. If so is this a redundant?

**Review by jphechter — COMMENTED** [review]
On the whole this looks good! 1 minor clarifying question regarding the test, but that's it.

File: bcdaworker/worker/alr_test.go (hunk 1 above)
Comment (katyasoup) [reply bcdaworker/worker/alr_test.go:127]: ooh good catch! Leftover from a merge - will remove the duplicate.

File: bcda/alr/gen/synthetic.go [hunk 2]
```diff
@@ -95,10 +100,21 @@ var valueGenerator = [...]regexpClosurePair{
...
+	{regexp.MustCompile("^COVID19_EPISODE$"), func() string { return strconv.Itoa(randomdata.Number(2)) }},
+	{regexp.MustCompile("^COVID19_MONTH(0[1-9]|1[0-2])$"), func() string { return strconv.Itoa(randomdata.Number(2)) }},
+	{regexp.MustCompile("^(ADMISSION_DT)$"), func() string { return randomDate(minCovidDate, maxCovidDate) }},
+	{regexp.MustCompile("^(DISCHARGE_DT)$"), func() string {
+		return randomEmpty(less,
+			func() string { return randomDate(maxCovidDate, time.Now()) })
+	}},
+	{regexp.MustCompile("^(U071)|(B9729)$"), func() string {
+		return randomEmpty(less,
+			func() string { return randomStringOption([]string{"U071", "B9729"}) })
```
Comment (CosmicSyntax) [inline bcda/alr/gen/synthetic.go:112]: I think the value returned is either 0 or 1.

Comment (CosmicSyntax) [reply bcda/alr/gen/synthetic.go:112]: Unless Bill said something else about this field.

Comment (katyasoup) [reply bcda/alr/gen/synthetic.go:112]: Based on the latest version of his IG, the diagnosis field should look like:
```  "diagnosis" : [
    {
      "condition" : {
        "identifier" : {
          "system" : "http://hl7.org/fhir/sid/icd-10",
          "value" : "U071"
        }
      }
    }
  ],```

BUT, it does need to be adjusted to only be included if the value is 1. Will adjust!

**Review by jphechter — APPROVED** [review]
LGTM 👍

**Review by SMLuthi — APPROVED** [review]
LGTM!

File: bcda/models/fhir/alr/v1/covidepisode.go [hunk 3]
```diff
@@ -128,21 +129,24 @@ func covidEpisode(mbi string, keyValue []utils.KvPair, lastUpdated time.Time) *f
...
+			if kv.Value != "0" {
+				episodeDiagnosis := []*fhirmodels.EpisodeOfCare_Diagnosis{}
+				diagnosis := &fhirmodels.EpisodeOfCare_Diagnosis{
+					Condition: &fhirdatatypes.Reference{
+						Identifier: &fhirdatatypes.Identifier{
+							System: &fhirdatatypes.Uri{
+								Value: "http://hl7.org/fhir/sid/icd-10",
+							},
+							Value: &fhirdatatypes.String{
+								Value: kv.Value,
```
Comment (CosmicSyntax) [inline bcda/models/fhir/alr/v1/covidepisode.go:142]: Should this be kv.Key?

File: bcda/models/fhir/alr/v2/covidepisode.go [hunk 4]
```diff
@@ -110,32 +111,42 @@ func covidEpisode(mbi string, keyValue []utils.KvPair, lastUpdated time.Time) *r
...
+			if kv.Value != "0" {
+				episodeDiagnosis := []*r4Models.EpisodeOfCare_Diagnosis{}
+				diagnosis := &r4Models.EpisodeOfCare_Diagnosis{
+					Condition: &r4Datatypes.Reference{
+						Identifier: &r4Datatypes.Identifier{
+							System: &r4Datatypes.Uri{
+								Value: "http://hl7.org/fhir/sid/icd-10",
+							},
+							Value: &r4Datatypes.String{
+								Value: kv.Value,
```
Comment (CosmicSyntax) [inline bcda/models/fhir/alr/v2/covidepisode.go:142]: Same as above?

**Review by CosmicSyntax — APPROVED** [review]
🎉

**Pull request merged**